pkg/config/mesh/networks_watcher.go

```diff
@@ -105,7 +115,7 @@ func (w *internalNetworkWatcher) PrevNetworks() *meshconfig.MeshNetworks {
 
 // SetNetworks will use the given value for mesh networks and notify all handlers of the change
 func (w *internalNetworkWatcher) SetNetworks(meshNetworks *meshconfig.MeshNetworks) {
-	var handlers []func()
+	var handlers []*WatcherHandler
```

Comment:
I think that if you'll changes handlers field to be `sets.Set` it will simplify the logic later on (e.g. in `DeleteNetworksHandler`) for lookup, insert and deletion of handlers.

Reply:
Yes. But the length of handlers is very small in most cases. So using slice may be enough.

Reply:
+1 to using slice, not set. Set will make ordering tricky and not relevant from performance POV here.

Reply:
Yes order matters here